### PR TITLE
React events and state

### PR DESCRIPTION
**Pull request by Johnny-Baltazar**
click-me-thank-you

### CODE REVIEW

**Comment by Johnny-Baltazar** [comment]
Somehow my react-class-components exercise also got added with this one.

File: react-events-and-state/src/index.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    this.setState(prevState => ({
+      isClicked: !prevState.isClicked
+    }));
+  }
+
+
+
+  render() {
+    if (this.state.isClicked) {
+      this.props.text = 'Thank you!';
```
Comment (scbowler) [inline react-events-and-state/src/index.jsx]: `this.props` is read only meaning we should never directly set any properties on it. Here it is better to use a variable:
```JSX
let text = '';

if (this.state.isClicked) {
  text = 'Thank you!';
} else {
  text = 'Click Me!';
}
```

File: react-events-and-state/src/index.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    } else {
+      this.props.text = 'Click Me!'
+    }
+
+    return(
+      <div>
+        <button onClick={this.handleClick}>
+          {this.props.text}
```
Comment (scbowler) [inline react-events-and-state/src/index.jsx]: Here should use the `text` variable
```JSX
{ text }
```

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good, but a couple updates needed. Let me know if you have any questions.

**Review by scbowler — APPROVED** [review]
Great job implementing the feedback, looks good! 👍

**Pull request merged**